File: exercicios_2_C/11_dobro_fn.c
/*
11) Escreva um programa que leia um número inteiro do usuário e retorne o seu dobro utilizando 
uma função.
*/ 

#include <stdio.h> // Inclui a biblioteca padrão de entrada e saída.

// Função para calcular o dobro de um número inteiro.
int calcularDobro(int numero) {
    return numero * 2; // Retorna o dobro do número fornecido.
}

int main() { // Função principal.
    int num, resultado; // Variáveis para armazenar o número fornecido pelo usuário e o resultado do cálculo.

    printf("Digite um número inteiro: "); // Solicita ao usuário que digite um número inteiro.
    scanf("%d", &num); // Lê e armazena o número fornecido pelo usuário.

    resultado = calcularDobro(num); // Chama a função 'calcularDobro' para calcular o dobro do número fornecido.

    printf("O dobro de %d é: %d\n", num, resultado); // Imprime o dobro do número fornecido.

    return 0; // Indica que o programa foi executado com sucesso.
}


/*
saída:
Digite um número inteiro: 8
O dobro de 8 é: 16
*/ 

File: exercicios_2_Rust/11_dobro_fn.rs
/*
11) Escreva um programa que leia um número inteiro do usuário e retorne o seu dobro utilizando 
uma função.
*/ 

use std::io;

fn calcular_dobro(numero: i32) -> i32 {
    numero * 2
}

fn main() {
    let mut input = String::new();

    println!("Digite um número inteiro: ");
    io::stdin().read_line(&mut input).expect("Falha ao ler a entrada");
    let num: i32 = input.trim().parse().expect("Entrada inválida");

    let resultado = calcular_dobro(num);

    println!("O dobro de {} é: {}", num, resultado);
}

/*
saída:
Digite um número inteiro: 8
O dobro de 8 é: 16
*/ 